lib/google_scraping/dashboard/keywords.ex

```diff
@@ -42,15 +43,73 @@ defmodule GoogleScraping.Dashboard.Keywords do
   Creates keywords list.
   """
   def create_keyword_list(keyword_list, user) do
-    Enum.each(keyword_list, fn keyword_name ->
-      keyword_params = %{
-        user_id: user.id,
-        name: keyword_name,
-        status: :new
-      }
-
-      # TODO: start background jobs for each created keyword
-      create_keyword(keyword_params)
+    keywords_data =
+      Enum.map(keyword_list, fn keyword_name ->
+        %{
+          user_id: user.id,
+          name: keyword_name,
+          status: :new
+        }
+      end)
+
+    Repo.transaction(fn ->
+      try do
+        keyword_ids = validate_and_create_keywords_list(keywords_data)
+
+        Enum.each(keyword_ids, fn keyword_id ->
+          create_keyword_background_job(keyword_id)
+        end)
+      catch
+        :error -> Repo.rollback(:invalid_keywords)
+      end
+    end)
+  end
+
+  @doc """
+  Gets a single keyword.
+  """
+  def get_keyword_by_id!(id), do: Repo.get!(Keyword, id)
+
+  @doc """
+  Mark keyword as in_progress
+  """
+  def mark_as_in_progress(keyword) do
```

Comment:
A minor comment but since we're using the bang function`update!` then let's name the function as a bang too.

This way we know it would raise an error, instead of returning a `{:error, etc}` tuple like some other functions 😄 

```suggestion
  def mark_as_in_progress!(keyword) do
```

Reply:
Done 😄 https://github.com/Nihisil/nimble-certification-elixir/pull/41/commits/b5fdf0bef1733d16780e5ad0c4df3f419a675946